src/main/java/com/example/controller/GraphController.java

```diff
@@ -158,6 +189,130 @@ public void initGraphView() {
         }
     }
 
+    private void saveGraphML(File file) throws IOException {
```

Comment:
I would create a new class that would be responsible for saving and reading files. Maybe something like GraphConverter?

Reply:
you need to pass the graph and vertex counter back and forth so it's probably less confusing to just leave those 2 functions inside

Reply:
We can remove vertex counter. I explained why in previous comments. So those methods can be extracted to new class. We will only pass file and graph as parameters if I'm not mistaken.